src/IECoreGL/MeshPrimitive.cpp

```diff
@@ -99,11 +99,11 @@ void MeshPrimitive::addPrimitiveVariable( const std::string &name, const IECore:
 
 	if ( primVar.interpolation==IECore::PrimitiveVariable::FaceVarying )
 	{
-		addVertexAttribute( name, primVar.data );
+		addVertexAttribute( name, primVar.expandedData() );
 	}
 	else if ( primVar.interpolation==IECore::PrimitiveVariable::Constant )
 	{
-		addUniformAttribute( name, primVar.data );
+		addUniformAttribute( name, primVar.expandedData() );
```

Comment:
Seems a little strange for us to allow indexed Constant primitive variables. Can this happen or we just being defensive as the API supports it now?

Reply:
I believe it happens already, for compressed string arrays coming from Houdini. That's not especially relevant here, but since the API allows it and it is trivial to support, I think it is is the right thing to do.